lib/msf/base/sessions/command_shell.rb

```diff
@@ -88,6 +88,21 @@ def shell_init
     return true
   end
 
+  def bootstrap(datastore = {}, handler = nil)
+    session = self
+
+    if datastore['AutoVerifySession']
+      token = Rex::Text.rand_text_alphanumeric(8..24)
+      response = shell_command("echo #{token}")
```

Comment:
Perhaps this method can be configured to accept a timeout like `shell_command_token`, too.